service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CloneControlledAzureStorageContainerResourceFlight.java

```diff
@@ -0,0 +1,125 @@
+    super(inputParameters, applicationContext);
+
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
+        JobMapKeys.AUTH_USER_INFO.getKeyName(),
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CLONING_INSTRUCTIONS,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID);
+
+    var flightBeanBag = FlightBeanBag.getFromObject(applicationContext);
+    var sourceResource =
+        inputParameters.get(WorkspaceFlightMapKeys.ResourceKeys.RESOURCE, ControlledResource.class);
+    var userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    var destinationWorkspaceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+
+    boolean mergePolicies =
+        Optional.ofNullable(
+                inputParameters.get(WorkspaceFlightMapKeys.MERGE_POLICIES, Boolean.class))
+            .orElse(false);
+    var cloningInstructions =
+        Optional.ofNullable(
+                inputParameters.get(
+                    WorkspaceFlightMapKeys.ControlledResourceKeys.CLONING_INSTRUCTIONS,
+                    CloningInstructions.class))
+            .orElse(sourceResource.getCloningInstructions());
+
+    final ControlledAzureStorageContainerResource sourceContainer =
+        sourceResource.castByEnum(WsmResourceType.CONTROLLED_AZURE_STORAGE_CONTAINER);
+    if (CloningInstructions.COPY_NOTHING == cloningInstructions) {
+      addStep(new SetNoOpContainerCloneResponseStep(sourceContainer));
+      return;
+    }
+
+    // Get the cloud context and store it in the working map
+    addStep(
+        new GetCloudContextStep(
+            destinationWorkspaceId,
+            CloudPlatform.AZURE,
+            flightBeanBag.getGcpCloudContextService(),
+            flightBeanBag.getAzureCloudContextService()),
+        RetryRules.shortDatabase());
+
+    // Flight plan
+    // 1. Check user has read access to source container
+    // 2. Gather controlled resource metadata for source object
+    // 3. Copy container definition to new container resource
+    // 3. If referenced, bail out as unsupported.
+    // 4. Copy files from source container to destination container (TODO)
+    addStep(
+        new CheckControlledResourceAuthStep(
+            sourceResource, flightBeanBag.getControlledResourceMetadataManager(), userRequest),
+        RetryRules.shortExponential());
+
+    if (mergePolicies) {
+      addStep(
+          new ClonePolicyAttributesStep(
+              sourceResource.getWorkspaceId(),
+              destinationWorkspaceId,
+              userRequest,
+              flightBeanBag.getTpsApiDispatch()));
+    }
+
+    addStep(
+        new RetrieveControlledResourceMetadataStep(
```

Comment:
Where does the retrieved metadata get stored?

Reply:
Ah, I think in the working map?

Reply:
Correct